File: backend/ruby_foundations/exercises/small_problems_101/easy_8/double_char_2.rb

def double_consonants(string)
  result = ''
  string.each_char do |char|
    case char
    when /[[a-z]&&[^aeiou]]/i
      result << char * 2
    else
      result << char
    end
  end
  result
end

puts double_consonants('String') == "SSttrrinngg"
puts double_consonants("Hello-World!") == "HHellllo-WWorrlldd!"
puts double_consonants("July 4th") == "JJullyy 4tthh"
puts double_consonants('') == ""

File: frontend/javascript/exercises/easy_5/double_char_2.js
function doubleConsonants(sentence) {
  var result = '';
  for (var i = 0; i < sentence.length; i++) {
    if (sentence[i].match(/[aeiou\d,.!-\s]/i)) {
      result += sentence[i];
    } else {
      result += sentence[i] + sentence[i];
    }
  }
  return result;
}

console.log(doubleConsonants('String'));       // "SSttrrinngg"
console.log(doubleConsonants('Hello-World!')); // "HHellllo-WWorrlldd!"
console.log(doubleConsonants('July 4th'));     // "JJullyy 4tthh"
console.log(doubleConsonants(''));             // ""